src/redux/slices/authSlice.js

```diff
@@ -1,21 +1,116 @@
 import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
 import axios from 'axios';
 
+/**
+ * requestLogin accepts an object { email, password } which is used to
+ * log the user in.
+ *
+ * The returned token is placed in both localStorage and the redux slice.
+ */
+export const requestLogin = createAsyncThunk(
+  'auth/requestLogin',
+  async (credentials, { rejectWithValue }) => {
+    try {
+      const { email, password } = credentials;
+      let { data } = await axios.post('http://localhost:3000/api/auth/login', {
+        email,
+        password,
+      });
+      localStorage.setItem('token', data.token);
+      return data;
+    } catch (err) {
+      console.error('error in requestLogin');
+      return rejectWithValue(err.message);
+    }
+  }
+);
+
+/**
+ * tryToken takes no parameters, but it does pull token from localStorage
+ * Once verified, the user info and token are placed in slice
+ */
+export const tryToken = createAsyncThunk(
+  'tryToken',
+  async (placeholder, { rejectWithValue }) => {
+    try {
+      console.log('hello from tryToken');
+      const token = localStorage.getItem('token');
+
+      if (!token) return {};
+
+      const { data } = await axios.get(`http://localhost:3000/api/auth`, {
```

Comment:
It may be good to add and env variable (me included) so that when we deploy we don't have to change all of these URLs

Reply:
good call...took a little bit of digging to figure that out (never done env vars on front end) -- referencing [this page](https://vitejs.dev/guide/env-and-mode.html), I put VITE_API_URL=http://localhost:3000 and update authSlice api calls to reflect. It should still work even without the .env setup (fails back to localhost:3000), but we'll all need to add it to our files and set it on deployed once that becomes a thing